File: projects/Mallard/ios/ExternalLink/ExternalLink.m
//
//  Ophan.m
//  RnAppWithKotlin
//
//  Created by Max Spencer on 25/06/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_REMAP_MODULE(RNExternalLink, ExternalLink, NSObject)

+ (BOOL)requiresMainQueueSetup
{
  return YES;
}

RCT_EXTERN_METHOD(open:
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(canOpen:
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)

@end

File: projects/Mallard/ios/ExternalLink/ExternalLink.swift
import Foundation
import StoreKit

@objc(ExternalLink)
class ExternalLink: NSObject {

  @objc(canOpen:resolver:rejecter:)
  func canOpen(resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject:RCTPromiseRejectBlock) -> Bool {
    do {
        let canOpen = ExternalLinkAccount.canOpen();
        resolve(canOpen);
    } catch let error {
      reject("Whoops - ios is sad", "really sad", nil)
    }
  }

  @objc(open:resolver:rejecter:)
  func open(resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject:RCTPromiseRejectBlock) -> Void {
    do {
        resolve(ExternalLinkAccount.open());
    } catch let error {
      reject("Whoops - ios is sad", "really sad", nil)
    }
  }
}
